### ISSUE TITLE
NS3005: Could not set argument.

### ISSUE THREAD

**Issue by voroninp** [nsubstitute/NSubstitute.Analyzers#223]
**Describe the bug**
I am aware about `Arg,Do<TArg>(...)` api, but I am curious why I see this analzyer warning:
> NS3005:Could not set argument.

This yields a warning:
![image](https://github.com/nsubstitute/NSubstitute/assets/9202028/2616bbae-cfb6-44b2-939e-af913d01962d)

But this does not:
![image](https://github.com/nsubstitute/NSubstitute/assets/9202028/d6785c6d-96d2-44d8-99de-652c38f8abed)

**To Reproduce**
```
TimerCallback? cb = null;
object? state = null;
var timer = Substitute.For<ITimer>();
var timeProvider = Substitute.For<TimeProvider>();
timeProvider.CreateTimer(Arg.Any<TimerCallback>(), Arg.Any<object?>(), Arg.Any<TimeSpan>(), Arg.Any<TimeSpan>())
    .Returns(timer)
    .AndDoes(ci =>
    {
        cb = ci[0] as TimerCallback;
        state = ci[1];
    });
```
**Expected behaviour**
There should not any warning here.

**Environment:**
 - NSubstitute version: 5.1.0
 - NSubstitute.Analyzers version: CSharp 1.0.17
 - Platform: [.net 8 on Windows]

**Comment by dtchepak** [comment]
Hi @voroninp ,

Thanks for raising this. I've transferred it to the Analyzers project which is detecting [NS3005](https://github.com/nsubstitute/NSubstitute.Analyzers/blob/dev/documentation/rules/NS3005.md).

cc: @tpodolak

**Comment by tpodolak** [comment]
@voroninp thanks for reporting. It loosk like analyzer confues 
```c#
 state = ci[1];
```
with
```c#
c[1] = state
```
